.github/workflows/ruby.yml

```diff
@@ -14,45 +14,93 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        os: [ ubuntu-18.04, macos, windows-latest ]
-        ruby: [ 2.3, 2.4, 2.5, 2.6, 2.7 ]
-        exclude:
-          - os: windows-latest
-            ruby: 2.3
+        os: [ ubuntu-18.04, macos ]
+        ruby: [ 2.3, 2.4, 2.5, 2.6, 2.7, ruby-head ]
```

Comment:
So, does this add ruby-head as a must-pass? Is that a good idea?

Reply:
> Is that a good idea?

What's Don think?  You don't really want to get me going on Ruby master builds, but...

Without pointing fingers, some of the previous builds used in CI were not fully tested, especially with the main MRI test suite 'test-all'.  So, a daily cron job would run and create a build based on a commit that didn't pass ruby/ruby CI.

All the builds used with current actions are fully tested.  If a build doesn't pass CI, the last passing build will be kept in place for CI use.

Hence, I think it's a risk worth taking.  If it breaks, ping me.  JFYI, that's what I've been working on for while, Windows builds (converting to Actions, adding mswin, etc)  and custom actions scripts.



> does this add ruby-head as a must-pass?

Due to the Actions UI, etc, there isn't a worthwhile way to do 'allow failure'.

Reply:
Oh, interesting. So it's not _really_ ruby-head, it's actually "the last ruby trunk build that passed it's own CI?"

Reply:
Correct.  If you can make up a good abbreviation for that, be happy to use it.

Since ruby/ruby switched to git, trunk is no more, although it might be an alias just because...

Reply:
Cool. I’m still not sure I want to make uncaught bugs on Ruby head crash our build and make it fail until they’re fixed...

Reply:
How about we call it 'master'?

> I want to make uncaught bugs on Ruby head crash

I think that's unlikely.  It would certainly be a good idea to test master sometime in Q4.  But if you're going to do that, might as well go year round.

I've been building/using master for 3+ years, and 'uncaught bugs' are very rare.  Puma is an important gem, and if it exposes a bug in master, isn't it better that they be identified before a Ruby release in December?

In the unlikely event that it does cause CI failures here, it's only two line of code to change...

Reply:
We'll try it, but I reserve the right to rollback 😆

Reply:
Please ping me first on any failures.  Thanks.